django_x509/admin.py

```diff
@@ -12,5 +22,13 @@ class CaAdmin(AbstractCaAdmin):
     pass
 
 
-admin.site.register(Ca, CaAdmin)
-admin.site.register(Cert, CertAdmin)
+# Check if model is registered, if not, register it
+# on admin. The check is required before registration
+# because this file is imported in tests/test_admin.py
+# which used in extended apps as well. The absense of
+# the check will cause `AlreadyRegistered` Error in
+# extended app during testing.
+if not admin.site.is_registered(Ca):
+    admin.site.register(Ca, CaAdmin)
+if not admin.site.is_registered(Cert):
+    admin.site.register(Cert, CertAdmin)
```

Comment:
Since tests needs admin, I added this. :smile:

Reply:
I have a doubt regarding this:

standard case: using django-x509
when the admin will be loaded, the models won't be registered yet, so admin.site.is_registered() will always return False.

extension case: openwisp-controller or other custom apps using this library
when the admin of openwisp-controller will be loaded, the models won't be registered yet, so admin.site.is_registered() will always return False.

So what's the use of this? It's like as if it's not there.

If we really want to do this, we should do:

```python
if 'django_x509' in settings.INSTALLED_APPS:
    # register
```

Reply:
It becomes useful for running tests in sample app when the user creates their own admin class, say they make `sample_x509.CaAdmin` and register it, i.e `Ca` is registered with `sample_x509.CaAdmin`.
Now, When running tests, in `tests/test_admin.py` contains line `from ..admin import CaAdmin, CertAdmin` which calls the `django_x509/admin.py` file, since it contains: `admin.site.register(Ca, CaAdmin)`; it tries to register `Ca` with `django_x509.CaAdmin`.
This raises: `django.contrib.admin.sites.AlreadyRegistered: The model Ca is already registered with 'sample_x509.CaAdmin'.`

To avoid this, I added this check, to check for `Ca` being already registered, if it is, we don't try to register again.

In all other cases, the statements are doing nothing.

If I use `if 'django_x509' in settings.INSTALLED_APPS:`; it wouldn't register CaAdmin & CertAdmin during normal use, right?

An alternative solution can be: `admin_site._registry[MODEL_NAME]` to get the admin class instance, I wanted to avoid this because `_registry` doesn't seem to be intended to be used like that.

Reply:
Example: it fails here: https://travis-ci.org/github/openwisp/django-x509/jobs/689742702#L369

Reply:
I think the solution of using `admin_site._registry[MODEL_NAME]` is valid and suited for our use case. We're going deep down the rabbit hole, so it's ok to use private methods and attributes of django if we know what we're doing and accept the risk that we'll have to change our code when django is updated

Reply:
We are going some rabbit hole! :smile:
Implemented using admin.site._registry. :smile: